internal/ast/expression_test.go

```diff
@@ -0,0 +1,116 @@
+package ast_test
+
+import (
+	"errors"
+	"testing"
+
+	"github.com/google/go-cmp/cmp"
+	"github.com/stretchr/testify/require"
+	"github.com/woven-planet/go-zserio/internal/ast"
+	"github.com/woven-planet/go-zserio/internal/parser"
+)
+
+func TestStringLiteralExpressions(t *testing.T) {
+	tests := map[string]struct {
+		testExpression *ast.Expression
+		expectedValue  string
+		expectedError  error
+	}{
+		"simple-string": {
+			testExpression: &ast.Expression{
+				Type: parser.ZserioParserSTRING_LITERAL,
+				Text: "\"TestString\"",
+			},
+			expectedValue: "TestString",
+		},
+		"empty-string": {
+			testExpression: &ast.Expression{
+				Type: parser.ZserioParserSTRING_LITERAL,
+				Text: "\"\"",
+			},
+			expectedValue: "",
+		},
+		"invalid-string": {
+			testExpression: &ast.Expression{
+				Type: parser.ZserioParserSTRING_LITERAL,
+				Text: "\"",
+			},
+			expectedError: errors.New("evaluate &{Type:101 Text:\" Operand1:<nil> Operand2:<nil> Operand3:<nil> ResultType: EvaluationState:1 ResultIntValue:0 ResultFloatValue:0 ResultStringValue: ResultBoolValue:false ResultSymbol:<nil> FullyResolved:true}: string literal should contain quotes"),
+		},
+	}
+	for name, test := range tests {
+		t.Run(name, func(t *testing.T) {
+			scope := &ast.Package{}
+			err := test.testExpression.Evaluate(scope)
+			if test.expectedError != nil {
+				require.NotNil(t, err)
+				if !cmp.Equal(test.expectedError.Error(), err.Error()) {
+					t.Errorf("result error doesn't match (expected: %v, actual: %v", test.expectedError, err)
+				}
+			} else {
+				require.Equal(t, ast.ExpressionTypeString, test.testExpression.ResultType)
+				require.Equal(t, ast.EvaluationStateComplete, test.testExpression.EvaluationState)
+				if diff := cmp.Diff(test.expectedValue, test.testExpression.ResultStringValue); diff != "" {
+					t.Errorf("do not match: (-expected +got)\n%s", diff)
+				}
+			}
```

Comment:
nit:
```suggestion
				assert.EqualError(t, err.Error(), test.expectedError.Error())
				return
			}
			require.Equal(t, ast.ExpressionTypeString, test.testExpression.ResultType)
			require.Equal(t, ast.EvaluationStateComplete, test.testExpression.EvaluationState)
			assert.Equal(t, test.expectedValue, test.testExpression.ResultStringValue)
```

The error equality could use the above function and we can make an early return. We could also cleanup the assertions.